frontend/src/features/mapSideBar/SidebarContents/AssociatedLandForm.tsx

```diff
@@ -0,0 +1,342 @@
+  const keycloak = useKeycloakWrapper();
+  const dispatch = useDispatch();
+  const api = useApi();
+  let initialValues = {
+    activeStep: 0,
+    activeTab: 0,
+    data: getInitialValues(),
+  };
+
+  initialValues.data.agencyId = keycloak.agencyId;
+
+  /**
+   * Combines yup validation with manual validation of financial data for performance reasons.
+   * Large forms can take 3-4 seconds to validate with an all-yup validation schema.
+   * This validation is significantly faster.
+   * @param values formik form values to validate.
+   */
+  const handleValidate = async (values: ISteppedFormValues<IFormParcel>) => {
+    const yupErrors: any = ParcelSchema.validate(values.data, { abortEarly: false }).then(
+      () => ({}),
+      (err: any) => yupToFormErrors(err),
+    );
+
+    let pidDuplicated = false;
+    if (values.data.pid && initialValues.data.pid !== values.data.pid) {
+      pidDuplicated = !(await isPidAvailable(values.data));
+    }
+
+    let pinDuplicated = false;
+    if (
+      values.data.pin &&
+      initialValues.data.pin !== values.data.pin &&
+      values.data.pin.toString().length < 10
+    ) {
+      pinDuplicated = !(await isPinAvailable(values.data));
+    }
+
+    let errors = await yupErrors;
+    const { buildings: buildingErrors, tabs, ...parcelErrors } = errors;
+    if (buildingErrors?.length) {
+      errors = { ...errors, tabs: [...(errors.tabs ?? []), ParcelDetailTabs.buildings] };
+    }
+    if (parcelErrors && Object.keys(parcelErrors).length) {
+      errors = { ...errors, tabs: [...(errors.tabs ?? []), ParcelDetailTabs.parcel] };
+    }
+    if (pidDuplicated) {
+      errors = { ...errors, pid: 'This PID is already in use.' };
+    }
+    if (pinDuplicated) {
+      errors = { ...errors, pin: 'This PIN is already in use.' };
+    }
+    return Promise.resolve(errors);
+  };
+
+  const isPidAvailable = async (values: IFormParcel): Promise<boolean> => {
+    const response = await api.isPidAvailable(values.id, values.pid);
+    return response?.available;
+  };
+
+  const isPinAvailable = async (values: IFormParcel): Promise<boolean> => {
+    const response = await api.isPinAvailable(values.id, values.pin);
+    return response?.available;
+  };
+  const formikParcels = props.formikRef?.current?.values?.data?.parcels;
+
+  return (
+    <Container className="landForm">
+      <SteppedForm
+        // Provide the steps
+        steps={[
+          { route: 'ownership', title: 'Land Ownership', completed: false, canGoToStep: true },
+          { route: 'identification', title: 'Parcel ID', completed: false, canGoToStep: true },
+          { route: 'usage', title: 'Usage', completed: false, canGoToStep: true },
+          { route: 'valuation', title: 'Valuation', completed: false, canGoToStep: true },
+          { route: 'review', title: 'Review', completed: false, canGoToStep: true },
+        ]}
+        getTabs={() => {
```

Comment:
The steppedform asks the caller for the method to get the most recent tabs. In this case they are built based off of the parcel list within formik.